### ISSUE TITLE
Don't write more code than needed

### ISSUE THREAD

**Issue by qborreda** [misterbl/star-wars-directory#2]
https://github.com/misterbl/star-wars-directory/blob/a760df28e6efb05ceae6a46bf4b4ed9e9e345d12/src/actions/thunks/getSingleInfo.ts#L24

Sometimes we don't have time to plan ahead or are too into the flow that we forget we can simplify things. the thunk would always set it to `loading`, so you can extract the `dispatch(fetchingSingleInfo(true));` out of the try/catch, and it would always set it back to false afterwards, `dispatch(fetchingSingleInfo(false));`. There is also no need for repetition on lines 24 & 27

**Comment by laurent-bouhnik-gh** [comment]
I totally agree with that, since then I have actually implemented sgas and moved this action outside the try catch block
